src/components/UserPhotoUploader/UserPhotoUploader.tsx

```diff
@@ -0,0 +1,118 @@
+'use client'
+import React, { useEffect, useState } from 'react'
+import styles from './UserPhotoUploader.module.scss'
+import Image from 'next/image'
+import { Button } from '../Button/Button'
+import { useSession } from 'next-auth/react'
+import { apiClient } from '@/lib/apiClient'
+import { ProfileModel } from '@/data/frontend/profile/types'
+import { useUploadContext } from '@/contexts/UploadContext'
+import { ErrorIcon } from '@/assets/icons/ErrorIcon'
+
+interface UserPhotoUploaderProps {
+  profile: ProfileModel | null
+}
+
+export const UserPhotoUploader = ({ profile }: UserPhotoUploaderProps) => {
+  const { data: session } = useSession()
+  const [selectedFile, setSelectedFile] = useState<File | null>(null)
+  const [userImage, setUserImage] = useState(
+    profile?.avatarUrl || session?.user.image,
+  )
+  const { triggerUpload, setTriggerUpload } = useUploadContext()
+  const [showErrorMessage, setShowErrorMessage] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (triggerUpload) {
+      handleUpload()
+      setTriggerUpload(false)
+    }
+  }, [triggerUpload])
+
+  const fetchUserAvatar = async () => {
+    try {
+      const avatarUrl = await apiClient.getUserAvatar()
+      setUserImage(avatarUrl)
+    } catch (error) {
+      console.error('Failed to fetch user avatar:', error)
+    }
+  }
+
+  useEffect(() => {
+    fetchUserAvatar()
+  }, [])
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0]
+    if (file) {
```

Comment:
we should check if the file is an image also. I think we shouldn't accept other files here